internal/hierarchyconfig/reconciler.go

```diff
@@ -137,8 +140,14 @@ func (r *Reconciler) handleUnmanaged(ctx context.Context, log logr.Logger, nm st
 		controllerutil.RemoveFinalizer(inst, api.FinalizerHasSubnamespace)
 		inst.Status.Children = nil
 		stats.WriteHierConfig()
-		if err := r.Update(ctx, inst); err != nil {
-			log.Error(err, "while removing finalizers on unmanaged namespace")
+
+		err := r.Patch(ctx, inst, client.MergeFrom(baseInst))
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				log.Info("Singleton has been deleted")
+				return nil
+			}
+			log.Error(err, "while removing finalizers and children on unmanaged singleton")
```

Comment:
I wonder if this could be factored out, e.g.:

```
err := r.Patch(...)
if err != nil {
  return somelib.handleMaybeDeletedError(err, "Singleton", "while removing finalizers and children...")
}
```

wdyt? That would make the code cleaner throughout.